src/xcode/ENA/ENA/Source/Services/HealthCertificateValidationService/HealthCertificateValidationService.swift

```diff
@@ -172,4 +212,557 @@ final class HealthCertificateValidationService: HealthCertificateValidationProvi
+			// all rules has to be .open
+			Log.info("Successfully combined rules: \(private: combinedRuleValidations). Validation result is: validationOpen. Validation complete.", log: .vaccination)
+			completion(.success(.validationOpen(combinedRuleValidations)))
+		} else {
+			// At least one rule should contain now .fail
+			Log.info("Successfully combined rules: \(private: combinedRuleValidations). Validation result is: validationFailed. Validation complete.", log: .vaccination)
+			completion(.success(.validationFailed(combinedRuleValidations)))
+		}
+	}
+	
+	// MARK: - Acceptance Rules
+		
+	private func downloadAcceptanceRule(
+		completion: @escaping (Result<[Rule], HealthCertificateValidationError>) -> Void
+	) {
+		client.getDCCRules(
+			eTag: store.acceptanceRulesCache?.lastValidationRulesETag,
+			isFake: false,
+			ruleType: .acceptance,
+			completion: { [weak self] result in
+				guard let self = self else {
+					Log.error("Could not create strong self", log: .vaccination)
+					completion(.failure(.ACCEPTANCE_RULE_CLIENT_ERROR))
+					return
+				}
+				
+				switch result {
+				case let .success(packageDownloadResponse):
+					self.acceptanceRuleDownloadingSuccessHandler(
+						packageDownloadResponse: packageDownloadResponse,
+						completion: completion
+					)
+	
+				case let .failure(error):
+					self.acceptanceRuleDownloadingFailureHandler(
+						error: error,
+						completion: completion
+					)
+				}
+			}
+		)
+	}
+	
+	private func acceptanceRuleDownloadingSuccessHandler(
+		packageDownloadResponse: PackageDownloadResponse,
+		completion: @escaping (Result<[Rule], HealthCertificateValidationError>) -> Void
+	) {
+		Log.info("Successfully received acceptance rules package. Proceed with eTag verification...", log: .vaccination)
+
+		guard let eTag = packageDownloadResponse.etag else {
+			Log.error("ETag of package is missing. Return with failure.", log: .vaccination)
+			completion(.failure(.ACCEPTANCE_RULE_JSON_ARCHIVE_SIGNATURE_INVALID))
+			return
+		}
+		
+		Log.info("Successfully verified eTag. Proceed with package extraction...", log: .vaccination)
+				
+		guard !packageDownloadResponse.isEmpty,
+			  let sapDownloadedPackage = packageDownloadResponse.package else {
+			Log.error("PackageDownloadResponse is empty. Return with failure.", log: .vaccination)
+			completion(.failure(.ACCEPTANCE_RULE_JSON_ARCHIVE_FILE_MISSING))
+			return
+		}
+		Log.info("Successfully extracted sapDownloadedPackage. Proceed with package verification...", log: .vaccination)
+		
+		guard self.signatureVerifier.verify(sapDownloadedPackage) else {
+			Log.error("Verification of sapDownloadedPackage failed. Return with failure", log: .vaccination)
+			completion(.failure(.ACCEPTANCE_RULE_JSON_ARCHIVE_SIGNATURE_INVALID))
+			return
+		}
+		Log.info("Successfully verified sapDownloadedPackage. Proceed now with CBOR decoding...", log: .vaccination)
+		
+		
+		self.acceptanceRules(sapDownloadedPackage.bin, completion: { result in
```

Comment:
```suggestion
		extractAcceptanceRules(sapDownloadedPackage.bin, completion: { result in
```
Something like that?

Reply:
yep